### PR TITLE
cp: --preserve should keep xattr

### PR DESCRIPTION
**Pull request by sylvestre**
Should help with tests/cp/acl.sh

### CODE REVIEW

File: tests/by-util/test_cp.rs [hunk 1]
```diff
@@ -3701,3 +3703,57 @@ fn test_cp_no_such() {
         .fails()
         .stderr_is("cp: 'no-such/' is not a directory\n");
 }
+
+#[cfg(all(unix, not(target_os = "macos")))]
+fn compare_xattrs<P: AsRef<Path>>(path1: P, path2: P) -> bool {
```
Comment (sylvestre) [inline tests/by-util/test_cp.rs:3708]: i will update this code to use the one in
https://github.com/uutils/coreutils/pull/5835

after this one is merged

**Comment by cakebaker** [comment]
> Congrats! The gnu test tests/cp/acl is no longer failing!

Kudos :)

File: tests/by-util/test_cp.rs [hunk 2]
```diff
@@ -3701,3 +3703,57 @@ fn test_cp_no_such() {
...
+    let file_target = "b/file";
+    at.mkdir(path1);
+    at.mkdir(path2);
+    at.touch(file);
+
+    let path = at.plus_as_string(file);
+    // calling the command directly. xattr requires some dev packages to be installed
+    // and it adds a complex dependency just for a test
+    match Command::new("setfacl")
+        .args(["-m", "group::rwx", &file])
```
Comment (cakebaker) [inline tests/by-util/test_cp.rs:3741]: There is a `setfacl: a/file: Not a directory` message when running the tests, so I guess it should be:
```suggestion
        .args(["-m", "group::rwx", &path1])
```

File: tests/by-util/test_cp.rs [hunk 3]
```diff
@@ -3701,3 +3703,57 @@ fn test_cp_no_such() {
         .fails()
         .stderr_is("cp: 'no-such/' is not a directory\n");
 }
+
+#[cfg(all(unix, not(target_os = "macos")))]
```
Comment (cakebaker) [inline tests/by-util/test_cp.rs:3707]: I think it should be the same as for `test_acl_preserve`:
```suggestion
#[cfg(all(unix, not(any(target_os = "android", target_os = "macos"))))]
```

File: tests/by-util/test_cp.rs
Comment (cakebaker) [inline tests/by-util/test_cp.rs:1]: Is it the expected behavior that the test doesn't fail if the changes in `cp.rs` are not applied?

File: tests/by-util/test_cp.rs [hunk 4]
```diff
@@ -56,6 +56,8 @@ static TEST_MOUNT_MOUNTPOINT: &str = "mount";
 static TEST_MOUNT_OTHER_FILESYSTEM_FILE: &str = "mount/DO_NOT_copy_me.txt";
 #[cfg(unix)]
 static TEST_NONEXISTENT_FILE: &str = "nonexistent_file.txt";
+#[cfg(all(unix, not(target_os = "macos")))]
```
Comment (cakebaker) [inline tests/by-util/test_cp.rs:59]: See my comment below.
```suggestion
#[cfg(all(unix, not(any(target_os = "android", target_os = "macos"))))]
```

File: tests/by-util/test_cp.rs
Comment (sylvestre) [reply tests/by-util/test_cp.rs:1]: yeah, i am planning to fix mac and the tests once they land :)

**Pull request merged**